Mathlib/Computability/DFA.lean

```diff
@@ -163,7 +174,50 @@ theorem pumping_lemma [Fintype σ] {x : List α} (hx : x ∈ M.accepts)
   rw [Set.mem_singleton_iff] at ha' hc'
   substs ha' hc'
   have h := M.evalFrom_of_pow hb hb'
-  rwa [mem_accepts, evalFrom_of_append, evalFrom_of_append, h, hc]
+  rwa [mem_accepts', evalFrom_of_append, evalFrom_of_append, h, hc]
 #align DFA.pumping_lemma DFA.pumping_lemma
 
+section Maps
+
+universe u' v'
+variable {α' : Type u'} {σ' : Type v'} (f : α ≃ α') (g : σ ≃ σ') (s : σ') (a : α') (x : List α')
+
+/-- Transforms the alphabet and states of a DFA. -/
+@[simps]
+def map (f : α' → α) (g : σ → σ') (h : σ' → σ) : DFA α' σ' where
+  step s a := g (M.step (h s) (f a))
+  start := g M.start
+  accept := g '' M.accept
+
+/-- Lifts equivalences on the alphabet and states to an equivalence on DFAs. -/
+def equiv (f : α ≃ α') (g : σ ≃ σ') : DFA α σ ≃ DFA α' σ' where
+  toFun M := M.map f.symm g g.symm
+  invFun M := M.map f g.symm g
+  left_inv M := by simp [map]
+  right_inv M := by simp [map]
+
```

Comment:
Can you add a lemma about `equiv refl refl` and `(equiv e f).symm = equiv e.symm f.symm`?